src/tiled/flipmapobjects.cpp

```diff
@@ -39,14 +40,39 @@ FlipMapObjects::FlipMapObjects(MapDocument *mapDocument,
     setText(QCoreApplication::translate("Undo Commands",
                                         "Flip %n Object(s)",
                                         nullptr, mapObjects.size()));
+
+    //computing objects center
+    QRectF boundaryObjectsRect;
+    for (MapObject *object : mMapObjects) {
```

Comment:
It is important to use `mapObjects` here rather than `mMapObjects`, because the latter is not `const`. Iterating a non-const Qt container that has implicit copies (as is the case here) will cause it to detach (make a deep copy).